vnext/local-cli/runWindows/utils/autolink.js

```diff
@@ -6,95 +6,359 @@
+        `Calculating ${chalk.bold(path.basename(csFile))}...`,
+        verbose,
       );
-      return;
+
+      const csContents = `// AutolinkedNativeModules.g.cs contents generated by "react-native autolink-windows"
+
+using System.Collections.Generic;${csUsingNamespaces}
+
+namespace Microsoft.ReactNative.Managed
+{
+    internal static class AutolinkedNativeModules
+    {
+        internal static void RegisterAutolinkedNativeModulePackages(IList<IReactPackageProvider> packageProviders)
+        {${csReactPacakgeProviders}
+        }
     }
+}
+`;
+
+      changesNecessary =
+        changesNecessary ||
+        updateFile(csFile, '\uFEFF' + csContents, verbose, checkMode);
+    } else if (projectLang === 'cpp') {
+      let cppIncludes = '';
+      let cppPackageProviders = '';
+
+      for (const dependencyName in windowsDependencies) {
+        windowsDependencies[dependencyName].projects.forEach(project => {
+          cppIncludes += `\n\n// Includes from ${dependencyName}`;
+          project.cppHeaders.forEach(header => {
+            cppIncludes += `\n#include <${header}>`;
+          });
 
-    //TODO: Update to the new RegisterAutolinkedNativeModulePackages method
-
-    let generatedIncludes =
-      '// AutolinkedNativeModules.g.h -- contents generated by "react-native run-windows"\\\r\n\\\r\n#pragma once\\\r\n\\\r\n';
-    const dependencies = config.dependencies;
-    let packageRegistrations =
-      '#define REGISTER_AUTOLINKED_NATIVE_MODULE_PACKAGES()';
-    for (const dependency in dependencies) {
-      const windowDependency = dependencies[dependency].platforms.windows;
-      const cppProjFile = windowDependency
-        ? windowDependency.cppProjFile
-        : null;
-      if (cppProjFile == null) {
-        console.log('No C++/WinRT project found for ' + dependency);
-        continue;
+          cppPackageProviders += `\n    // IReactPackageProviders from ${dependencyName}`;
+          project.cppPackageProviders.forEach(packageProvider => {
+            cppPackageProviders += `\n    packageProviders.Append(winrt::${packageProvider}());`;
+          });
+        });
       }
-      console.log(
-        `Adding include and package provider statement for ${dependency}`,
+
+      const cppHeaderFile = path.join(
+        projectDir,
+        'AutolinkedNativeModules.g.h',
+      );
+
+      verboseMessage(
+        `Calculating ${chalk.bold(path.basename(cppHeaderFile))}...`,
+        verbose,
       );
-      const trimmedPackageName = windowDependency.packageName.trim();
-      const includeStatement = `\r\n#include <winrt/${trimmedPackageName}.h>`;
-      generatedIncludes += includeStatement;
-      const packageRegistration = `    \\\r\n    PackageProviders().Append(winrt::${trimmedPackageName}::ReactPackageProvider());`;
-      packageRegistrations += packageRegistration;
+
+      const cppHeaderContents = `// AutolinkedNativeModules.g.h contents generated by "react-native autolink-windows"
```

Comment:
Do we want to disable Clang format for this as well?

Reply:
We need to disable the clang formatting in the template files so our repo formatting settings don't touch it. But the CLI actually removes that line from the file when it creates a new app, because it's not for us to decide if an app repo wants to use clang, or not, and what formatting settings they use.